internal/provider/app_data_source.go

```diff
@@ -96,23 +98,21 @@ func (d appDataSource) Read(ctx context.Context, req datasource.ReadRequest, res
 		return
 	}
 
-	appName := data.Name.Value
+	appName := data.Name.ValueString()
 
-	queryresp, err := graphql.GetFullApp(context.Background(), *d.provider.client, appName)
+	queryresp, err := graphql.GetFullApp(context.Background(), *d.client, appName)
```

Comment:
You should pass the passed `ctx` instead making a new context by calling context.Background().

Reply:
I noticed these as well, but wasn't sure if I should try to limit the scope of the PR to just what was required to move to the new framework version, because I wasn't certain what the effects of changing the context would be.